openmetadata-ui/src/main/resources/ui/src/components/common/FormBuilder/FormBuilder.tsx

```diff
@@ -152,7 +154,136 @@ const FormBuilder: FunctionComponent<Props> = forwardRef(
         onChange={handleFormChange}
         onFocus={onFocus}
         onSubmit={onSubmit}
-        {...props}>
+        {...props}
+        validator={{
+          ...props?.validator,
+          validateFormData: (formData: ConfigData) => {
+            const validationErrors: Record<string, { __errors: string[] }> = {};
+
+            const resolveRef = (ref: string, schemaObj: any) => {
+              if (!ref) {
+                return null;
+              }
+              const refPath = ref.split('/').slice(1);
+
+              return refPath.reduce((obj, path) => obj?.[path], schemaObj);
+            };
+
+            const validateProperty = (
+              property: any,
+              value: any,
+              key: string,
+              path: string
+            ): { __errors: string[] } | null => {
+              // Handle schema references
+              if (property.$ref) {
+                const refSchema = resolveRef(property.$ref, schema);
+                if (!refSchema) {
+                  return null;
+                }
+
+                return validateProperty(refSchema, value, key, path);
+              }
+
+              if (
+                typeof property === 'object' &&
+                property !== null &&
```

Comment:
where ever we are checking the type, we can use lodash function there for better readability!